src/syrupy/types.py

```diff
@@ -1,8 +1,14 @@
 from typing import (
     Any,
+    Callable,
+    Hashable,
+    Optional,
+    Tuple,
     Union,
 )
 
 
 SerializableData = Any
 SerializedData = Union[str, bytes]
+PropertyPath = Tuple[Hashable, ...]
```

Comment:
Should types be included in the path sequence? i.e.
```suggestion
PropertyPath = Tuple[Tuple[Hashable, Type[SerializableData]], ...]
```

Reply:
Including types would allow for matching on the parent types

Reply:
@noahnu any thoughts on this?